buildstockbatch/cloud/docker_base.py

```diff
@@ -461,3 +462,44 @@ def run_simulations(cls, cfg, job_id, jobs_d, sim_dir, fs, output_path):
                 shutil.rmtree(item)
             elif os.path.isfile(item):
                 os.remove(item)
+
+    def log_summary(self):
+        """
+        Log a summary of how many simulations succeeded, failed, or ended with other statuses.
+
+        Uses the `completed_status` column of the files in results_csvs/results_*.csv.gz.
+        """
+        fs = self.get_fs()
+        # Summary of simulation statuses across all upgrades
+        status_summary = {}
+        all_statuses = set()
+
+        results_csv_dir = f"{self.results_dir}/results_csvs/"
+        try:
+            results_files = fs.ls(results_csv_dir)
+        except FileNotFoundError:
+            logger.info(f"No results CSV files found at {results_csv_dir}")
+            return
+
+        for result in results_files:
+            upgrade_id = result.split(".")[0][-2:]
+            with fs.open(result) as f:
+                with gzip.open(f) as gf:
+                    df = pd.read_csv(gf, usecols=["completed_status"])
+            # Dict mapping from status (e.g. "Success") to count
+            statuses = df.groupby("completed_status").size().to_dict()
+            status_summary[upgrade_id] = statuses
+            all_statuses.update(statuses.keys())
+
+        # Always include these statuses and show them first
+        always_use = ["Success", "Fail"]
```

Comment:
I think there is another category which is Invalid. So, would be helpful to add that too. That basically implies that upgrades is invalid for the baseline building